### PR TITLE
Reactivate xz

### PR DESCRIPTION
**Pull request by thesamesam**
Lasse and I have discussed this and we'd like oss-fuzz working again on the repository as fixes and various cleanups continue to be committed.

The malicious test files have been purged already in https://github.com/tukaani-project/xz/commit/e93e13c8b3bec925c56e0c0b675d8000a0f7f754.

Obviously will need an ACK from @Larhzu.

### CODE REVIEW

**Comment by thesamesam** [comment]
I have no idea about ae9dd26d624ec2cd2ae1d277d62ce30e2a70e798 fwiw, other than being confident that it's not caught up in this - i.e. I have no association with LZMA SDK / 7zip (totally distinct project from xz) development at all.

As such, if y'all want to handle it in a separate PR, that's fine, but I'm probably not the person to drive that. I don't know who should be contacted to get it re-activated though. As far as I'm concerned, it was a bystander victim here in the end (not saying the revert was unreasonable or anything, just where we are now).

Let me know what's best. Thanks!

**Comment by thesamesam** [comment]
cc @jonathanmetzman

**Comment by Larhzu** [comment]
I'm happy to let @thesamesam handle the fuzzing related communication. Thanks! :-)

File: projects/lzma/project.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+sanitizers:
+  - address
+  - memory
+  - undefined
+
+fuzzing_engines:
+  - afl
+  - honggfuzz
+  - libfuzzer
+
```
[the remark this answers is not in the document]
Comment (thesamesam) [reply projects/lzma/project.yaml:16]: A great question which I have no idea how to answer. LZMA SDK is definitely maintained upstream but it doesn't look like it's had any attention for oss-fuzz in the last few years.

Should I split the LZMA SDK parts into a separate PR, or leave it entirely? I don't have any association with that project, so not sure if I'm the best person to do that. I just felt bad that they got caught up in the crossfire.

@fancycode What do you think?

[the remark this answers is not in the document]
Comment (thesamesam) [reply projects/lzma/project.yaml:16]: And maybe: @Larhzu maybe we can ask Igor about it although not sure if he has a Google account..

[the remark this answers is not in the document]
Comment (thesamesam) [reply projects/lzma/project.yaml:16]: In the meantime, I think I should just drop those parts from this PR, as it seems kind of open-ended as I'm not sure what the outcome will be there.

[the remark this answers is not in the document]
Comment (fancycode) [reply projects/lzma/project.yaml]: > A great question which I have no idea how to answer. LZMA SDK is definitely maintained upstream but it doesn't look like it's had any attention for oss-fuzz in the last few years.
> 
> Should I split the LZMA SDK parts into a separate PR, or leave it entirely? I don't have any association with that project, so not sure if I'm the best person to do that. I just felt bad that they got caught up in the crossfire.
> 
> @fancycode What do you think?

@thesamesam Thanks for the heads up! Feel free to put the LZMA SDK parts in a separate PR. While this indeed was not maintained recently, I just started again, will upgrade to the latest SDK version in https://github.com/fancycode/lzma-fuzz/pull/4 and will try to keep this up to date better in the future. Having the SDK check integrated in CI should help.

**Comment by DonggeLiu** [comment]
Thanks @thesamesam.
Could you please address the header-check failure?

**Comment by thesamesam** [comment]
@DonggeLiu No problem, thanks for the review. I will do now. I wasn't sure if I should handle it given I was just reverting, but let's do it. Gimme a sec..

**Comment by thesamesam** [comment]
By the way, I wouldn't want it to block merging the reactivation, but if anyone is willing or able to help improve our fuzzing setup for xz, please do let us know / feel free to take a look. We really want people to be confident in xz and any/all suggestions are welcome.

File: projects/lzma/project.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (thesamesam) [reply projects/lzma/project.yaml]: Thanks! I opened up https://github.com/google/oss-fuzz/pull/11815.

**Review by DonggeLiu — APPROVED** [review]

**Pull request merged**